### PR TITLE
Docs: Add Accessibility guidelines page

### PR DESCRIPTION
**Pull request by ayeshakmaz**
### Links

- [Jira](https://jira.pinadmin.com/browse/PDS-2823)
- [TDD](link to Paper doc)
- [Figma](link to Figma file)

### Checklist

- [ ] Added unit and Flow Tests
- [ ] Added documentation + accessibility tests
- [ ] Verified accessibility: keyboard & screen reader interaction
- [ ] Checked dark mode, responsiveness, and right-to-left support
- [ ] Checked stakeholder feedback (e.g. Gestalt designers)

### CODE REVIEW

File: docs/src/Accessibility.doc.js [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+  <MainSection
+    name="Design considerations"
+    description="Accessibility starts at the design phase! Below are some key things to watch out for when designing inclusive products. For further detail and matching visual examples, check out our [Accessible Design deck](https://docs.google.com/presentation/d/1b-L0tuzaMTIf1xX7j86g46QfDW3_C0Ep_Ca4TEmXPz8/edit?usp=sharing)."
+  >
+    <MainSection.Subsection
+      title="Visuals"
+      description={`
+      When designing, it's important to ensure our use of color is appropriate.
+
+      - **Avoid using color as the sole indicator of information.** For instance, always supply and icon or text describing errors, rather than relying on a red outline or red text. Color-only changes do not work well for those who may be color blind or have low vision.
```
Comment (dangerismycat) [inline docs/src/Accessibility.doc.js:39]: typo: For instance, always supply **and** icon or text

File: docs/src/Accessibility.doc.js [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+    - **Create descriptive link text.** Avoid multiple links with vague text like "More info" or "Learn more". Instead, users should be able to tell exactly what info will be provided at a given link by the link text only. For instance, "Explore campaigns" instead of just "Learn more".
+    `}
+    />
+    <MainSection.Subsection
+      title="Responsiveness"
+      description={`
+    Accessibility requirements vary depending on the device used, so be sure to account for small and large screens alike.
+
+    - **Account for zooming!** Many users, especially on mobile, adjust the default size of fonts and often increase the zoom setting. Does your design still work when zoomed in 200%?
+    - **Create large touch targets.** If you've ever clicked on the wrong thing because the tap target was too small, you know how frustrating it can be to have to re-orient yourself. Ensure your touch targets on mobile are large enough for people with limited mobility to accurately press what they're aiming for.`}
```
Comment (dangerismycat) [inline docs/src/Accessibility.doc.js:68]: > Ensure your touch targets on mobile
I don't think this is only on mobile, right? I see tap targets that are way to small on desktop as well

Comment (ayeshakmaz) [reply docs/src/Accessibility.doc.js]: definitely applies to both, but is much more prevalent on mobile. I'll update

File: docs/src/Accessibility.doc.js [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+
+        Another key aspect to keyboard navigation is [focus management](https://css-tricks.com/focus-management-and-inert/). Users should never lose their place within a page, and their current point of focus should always be clear visually. Some components, like Modals or Popovers, act as focus traps, meaning the user's focus should only rotate between items inside the Modal or Popover, and anything below these items is not reachable by keyboard. When a component like Modal or Popover is dismissed or closed, the user's focus should go back to the item that triggered the component.
+    `}
+    />
+    <MainSection.Subsection
+      columns={2}
+      title="Available hooks"
+      description={`Gestalt provides two functions that help create more accessible experiences: \`useFocusVisible\` and \`useReducedMotion\`.`}
```
Comment (dangerismycat) [inline docs/src/Accessibility.doc.js:106]: We can add external links here, right? Consider replacing "functions" with "custom Hooks" and link that to the React docs on custom Hooks

File: docs/src/Accessibility.doc.js [hunk 4]
```diff
@@ -0,0 +1,250 @@
...
+          References:
+          <ul>
+            <li><a href="https://www.w3.org/WAI/WCAG21/Understanding/focus-visible.html">WCAG 2.4.7: Focus Visible</a></li>
+            <li><a href="https://developer.mozilla.org/en-US/docs/Web/CSS/:focus-visible">:focus-visible CSS pseudo-class</a></li>
+          </ul>
+        `}
+        defaultCode={`
+function Example() {
+  const { isFocusVisible } = useFocusVisible();
+  const [ focussedButton1, setFocussedButton1 ] = React.useState(false);
```
Comment (dangerismycat) [inline docs/src/Accessibility.doc.js:123]: nit: "focused" only has one 's'

File: docs/src/Accessibility.doc.js [hunk 5]
```diff
@@ -0,0 +1,250 @@
...
+      </Flex>
+    </Flex>
+  );
+}`}
+      />
+      <MainSection.Card
+        cardSize="md"
+        title="useReducedMotion"
+        description={`
+          \`useReducedMotion\` allows a user to request that the system minimize the amount of non-essential motion.
```
Comment (dangerismycat) [inline docs/src/Accessibility.doc.js:162]: I think this should perhaps be more direct and say that all animations should first use this hook, and then show a non-animated version if requested? What do you think?

File: docs/src/Accessibility.doc.js [hunk 6]
```diff
@@ -0,0 +1,250 @@
...
+        }`}
+      />
+    </MainSection.Subsection>
+  </MainSection>,
+);
+
+card(
+  <MainSection name="Further learning">
+    <MainSection.Subsection
+      title="Internal"
```
Comment (dangerismycat) [inline docs/src/Accessibility.doc.js:216]: "Pinternal"? 😅

**Review by dangerismycat — APPROVED** [review]

File: docs/src/Accessibility.doc.js (hunk 6 above)
Comment (ayeshakmaz) [reply docs/src/Accessibility.doc.js:216]: oh my gosh yes

File: docs/src/Accessibility.doc.js (hunk 5 above)
Comment (ayeshakmaz) [reply docs/src/Accessibility.doc.js:162]: I'm going to save this for when we take a full pass at the stand alone page, but yeah, we can probably update this part

**Pull request merged**